### PR TITLE
feat: disambiguation in mgt

### PR DESCRIPTION
**Pull request by gavinbarron**
adds documentation to cover the new disambiguation feature in MGT v3

### CODE REVIEW

**Review by waldekmastykarz — CHANGES_REQUESTED** [review]
Let's do a couple of cosmetic changes. Other than that, it looks awesome. Let's merge it!

File: concepts/toolkit/customize-components/disambiguation.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+To mitigate this challenge we built the [`mgt-spfx`](https://github.com/microsoftgraph/microsoft-graph-toolkit/tree/main/packages/mgt-spfx) package. Using `mgt-spfx` developers can centralize the registration of MGT web components across all SPFx solutions deployed on the tenant. By reusing MGT components from a central location web parts from different solutions can be loaded into a single page without throwing errors. When using `mgt-spfx` all MGT based web parts in a SharePoint tenant use the same version of MGT.
+
+To allow developers to build web parts using the latest version of MGT and load them on pages along with web parts that use v2.x of MGT, we've added a new disambiguation feature to MGT. Using this feature developers can specify a unique string to add to the tag name of all MGT web components in their application. When using disambiguation the suppied value is inserted as the second segment of the tag name, so when using `customElementHelper.withDisambiguation('foo)` the `<mgt-login>` tag is referenced using `<mgt-foo-login>`.
+
+## Usage in standard HTML and JavaScript
```
[the remark this answers is not in the document]
Comment (gavinbarron) [reply concepts/toolkit/customize-components/disambiguation.md:16]: Great suggestion!

File: concepts/toolkit/customize-components/disambiguation.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+</script>
+
+<mgt-contoso-login></mgt-contoso-login>
+<mgt-contoso-person person-query="Bill Gates" person-card="hover"></mgt-contoso-person>
+<mgt-contoso-agenda group-by-day></mgt-contoso-agenda>
+```
+
+> Note: the `import` of `mgt-components` must use a [dynamic import](#dynamic-imports-aka-lazy-loading) to ensure that the disambiguation is applied before the components are imported.
+
+
```
Comment (gavinbarron) [inline concepts/toolkit/customize-components/disambiguation.md:41]:
```suggestion

```

File: concepts/toolkit/customize-components/disambiguation.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  // import the components using dynamic import to avoid hoisting
+  import('@microsoft/mgt-components');
+</script>
+
+<mgt-contoso-login></mgt-contoso-login>
+<mgt-contoso-person person-query="Bill Gates" person-card="hover"></mgt-contoso-person>
+<mgt-contoso-agenda group-by-day></mgt-contoso-agenda>
+```
+
+> Note: the `import` of `mgt-components` must use a [dynamic import](#dynamic-imports-aka-lazy-loading) to ensure that the disambiguation is applied before the components are imported.
```
Comment (gavinbarron) [inline concepts/toolkit/customize-components/disambiguation.md:39]:
```suggestion
> [!Important]
> The `import` of `mgt-components` must use a [dynamic import](#dynamic-imports-aka-lazy-loading) to ensure that the disambiguation is applied before the components are imported. If a static import is used it is [hoisted](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting) and the import will occur before disambiguation can be applied.
```

File: concepts/toolkit/customize-components/disambiguation.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+<script type="module">
+  import { Providers, customElementHelper } from '@microsoft/mgt-element';
+  import { Msal2Provider } from '@microsoft/mgt-msal2-provider';
+  // configure disambiguation
+  customElementHelper.withDisambiguation('contoso');
+
+  // initialize the auth provider globally
+  Providers.globalProvider = new Msal2Provider({clientId: 'clientId'});
+
+  // import the components using dynamic import to avoid hoisting
```
[the remark this answers is not in the document]
Comment (gavinbarron) [reply concepts/toolkit/customize-components/disambiguation.md:30]: Added to the note below the code sample

File: concepts/toolkit/customize-components/disambiguation.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gavinbarron) [reply concepts/toolkit/customize-components/disambiguation.md]: Reversed the example usage sections

**Comment by gavinbarron** [comment]
@waldekmastykarz thanks for the suggestions, this is ready for another look from you.

**Pull request merged**